mars/dataframe/base/transpose.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 1999-2021 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+from ..operands import DataFrameOperand, DataFrameOperandMixin
+from ...core import OutputType
+from ..utils import parse_index
+from ...core.operand import OperandStage
+
+
+def reverse(x):
+    if x is None:
+        return
+    return x[::-1]
+
+
+class DataFrameTranspose(DataFrameOperand, DataFrameOperandMixin):
+
+    def __init__(self, **kw):
+        super().__init__(**kw)
+        self.output_types = [OutputType.dataframe]
+
+    def __call__(self, args):
+        arg = args[0]
+        new_shape = arg.shape[::-1]
+        dtypes = arg.dtypes
+        columns_value = arg.index_value
+        index_value = parse_index(dtypes.index)
+        return self.new_dataframe([arg], shape=new_shape, dtypes=dtypes,
+                                  columns_value=columns_value, index_value=index_value)
+
+    @classmethod
+    def tile(cls, op):
+        out_chunks = []
+        for c in op.inputs[0].chunks:
+            chunk_op = op.copy().reset_key()
+            chunk_op.stage = OperandStage.map
```

Comment:
`stage` is for those mapreduce operands as their execute functions are different for different stages, here transpose is not needed.